includes/admin/class-amp-post-meta-box.php

```diff
@@ -202,13 +279,99 @@ public function render_status( $post ) {
+	 * That string is in the 0 index of the array(), and the href is in the 1 index.
+	 *
+	 * @since 1.0
+	 * @param string $status The AMP enabled status.
+	 * @param array  $errors The AMP enabled errors.
+	 * @return array $error_messages The error message(s), each message being either a string or an array of strings.
+	 */
+	public function get_raw_error_messages( $status, $errors ) {
+		$error_messages = array();
+		if ( in_array( 'status_immutable', $errors, true ) ) {
+			if ( self::ENABLED_STATUS === $status ) {
+				$error_messages[] = __( 'Your site does not allow AMP to be disabled.', 'amp' );
+			} else {
+				$error_messages[] = __( 'Your site does not allow AMP to be enabled.', 'amp' );
+			}
+		}
+		if ( in_array( 'template_unsupported', $errors, true ) || in_array( 'no_matching_template', $errors, true ) ) {
+			$error_messages[] = array(
+				/* translators: %s is opening <a> for AMP settings screen, %s is closing </a> */
+				__( 'There are no %ssupported templates%s to display this in AMP.', 'amp' ), // phpcs:ignore WordPress.WP.I18n.UnorderedPlaceholdersText, WordPress.Arrays.ArrayDeclarationSpacing.ArrayItemNoNewLine
+				esc_url( admin_url( 'admin.php?page=' . AMP_Options_Manager::OPTION_NAME ) ),
+			);
+		}
+		if ( in_array( 'password-protected', $errors, true ) ) {
+			$error_messages[] = __( 'AMP cannot be enabled on password protected posts.', 'amp' );
+		}
+		if ( in_array( 'post-type-support', $errors, true ) ) {
+			$error_messages[] = array(
+				/* translators: %s is opening <a> for AMP settings screen, %s is closing </a> */
+				__( 'AMP cannot be enabled because this %spost type does not support it%s.', 'amp' ), // phpcs:ignore WordPress.WP.I18n.UnorderedPlaceholdersText, WordPress.Arrays.ArrayDeclarationSpacing.ArrayItemNoNewLine
+				esc_url( admin_url( 'admin.php?page=' . AMP_Options_Manager::OPTION_NAME ) ),
+			);
+		}
+		if ( in_array( 'skip-post', $errors, true ) ) {
+			$error_messages[] = __( 'A plugin or theme has disabled AMP support.', 'amp' );
+		}
+		if ( count( array_diff( $errors, array( 'status_immutable', 'page-on-front', 'page-for-posts', 'password-protected', 'post-type-support', 'skip-post', 'template_unsupported', 'no_matching_template' ) ) ) > 0 ) {
+			$error_messages[] = __( 'Unavailable for an unknown reason.', 'amp' );
+		}
+
+		return $error_messages;
+	}
+
+	/**
+	 * Gets the AMP enabled error message(s) for the Classic Editor.
+	 *
+	 * AMP can be disabled for several reasons,
+	 * like if the user disables it in Supported Templates > Content Types.
+	 * Then, it would not make sense to allow the user to enable AMP in the post editor,
+	 * as it's already disabled.
+	 * So this gets the message(s) that should display.
+	 *
+	 * @param array $raw_error_messages The raw error messages, possibly with URLs.
+	 * @return array $error_messages The error message(s), as an array of strings.
+	 */
+	public function get_classic_editor_error_messages( $raw_error_messages ) {
```

Comment:
There is duplication between PHP here and the JS in `PluginPostStatusInfo`. I appreciate the concern to not `dangerouslySetInnerHTML` but since the error messages are coming from a trusted source, I think it would be better to do so in this case. Even in Gutenberg, for example, there is a [`RawHTML`](https://github.com/WordPress/gutenberg/blob/master/packages/element/src/raw-html.js) component for this purpose. This component is publicly available already in Gutenberg via `wp.element.RawHTML`.

Reply:
Thanks, I'm working on this now.

Reply:
[These commits](https://github.com/Automattic/amp-wp/compare/9f0e248b5863563171294a8ce52cfb3c67f0c27e...230d87a42a2b0f73074736b0db11142a68ad6399) simplify the error messages to use `RawHTML` like you suggested.